File: 2021/src/test/java/de/jfriemel/aoc/days/Day06Test.java
package de.jfriemel.aoc.days;

import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.assertEquals;

public class Day06Test {

    private static final String INPUT_STRING = """
            3,4,3,1,2""";

    private static final List<String> INPUT = Arrays.asList(INPUT_STRING.split("\n"));

    @Test
    public void testPart1() {
        String result = new Day06().part1(INPUT);
        assertEquals("5934", result);
    }

    @Test
    public void testPart2() {
        String result = new Day06().part2(INPUT);
        assertEquals("26984457539", result);
    }

}

File: 2023/src/test/kotlin/de/jfriemel/aoc/days/Day06Test.kt
package de.jfriemel.aoc.days

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class Day06Test {
    private val input = """
        Time:      7  15   30
        Distance:  9  40  200
    """.trimIndent().lines()

    @Test
    fun testPart1() {
        val result = Day06.part1(input)
        Assertions.assertEquals("288", result)
    }

    @Test
    fun testPart2() {
        val result = Day06.part2(input)
        Assertions.assertEquals("71503", result)
    }
}
